includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2127,8 +2127,8 @@ public function get_payment_method_types( $payment_information ): array {
 			$order_id        = $order instanceof WC_Order ? $order->get_id() : null;
 			$payment_methods = $this->get_payment_methods_from_gateway_id( $token->get_gateway_id(), $order_id );
 		} else {
-			// Final fallback case, if all else fails.
-			$payment_methods = WC_Payments::get_gateway()->get_payment_method_ids_enabled_at_checkout( null, true );
+			// Final fallback case, if all else fails. WooPay payments will use this case.
+			$payment_methods = [ Payment_Method::CARD ];
```

Comment:
This fallback is general, and in case any code calling this method has some bug in the payment method area, we will fall to the card payment method, which doesn't seem ideal. 

Since it's WooPay and ECE, can we specify the fallback for them? 
I checked WooPay, can the following line 

```php
$woopay_intent_id = sanitize_user( wp_unslash( $_POST['platform-checkout-intent'] ?? '' ), true );
```

implemented [after the `get_payment_method_types` is called](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L1578) be added to the more specific check, rather than returning card if for everything? 

Regarding ECE, I tested and it seems that we have mappings ([#1](https://github.com/Automattic/woocommerce-payments/blob/develop/client/tokenized-express-checkout/transformers/stripe-to-wc.js#L76-L79) [#2](https://github.com/Automattic/woocommerce-payments/blob/develop/client/tokenized-payment-request/transformers/stripe-to-wc.js#L85-L88)) which will ensure the `payment_method` request param will be always present for this type of payment flow. Can we double-check whether this is the case and if yes then it seems that the only thing to fall back to is the WooPay scenario which can be specified by using the `platform-checkout-intent` payment request param, as mentioned above.